### PR TITLE
Fixes create and update goals docs

### LINKED ISSUE #37488: Quarkus Maven Plugin doc parameters are broken
### Describe the bug

Currently it uses the `name` which is not the actual parameter name as defined in the [doc](https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Parameter.html#name()) (name is when configuring from the pom.xml), it should use the `property` instead:
- https://github.com/quarkusio/quarkus/blob/main/devtools/maven/src/main/java/io/quarkus/maven/UpdateMojo.java#L72C4-L72C49
- https://github.com/quarkusio/quarkus/blob/main/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java#L105C20-L105C33

Because of this the doc parameters names a wrong:
- https://quarkus.io/guides/quarkus-maven-plugin#quarkus-maven-plugin-goal-update-rewriteupdaterecipesversion
- https://quarkus.io/guides/quarkus-maven-plugin#quarkus-maven-plugin-goal-add-extension-bomartifactid

We need to make sure the `@Parameter(property=...` is used in the doc when set (I would hope that the fieldName is used if not set).

### Expected behavior

Have the right parameter names in the doc

### Actual behavior

it uses the fieldName because `name` is not the actual name of the property at all, see:
https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Parameter.html#name()

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes #37488

### CODE REVIEW

**Comment by mcruzdev** [comment]
Hi @ia3andy, what do you think about this solution, I am not sure if this can cause a break change for plugin consumers. WYT?

**Comment by ia3andy** [comment]
@mcruzdev are those the only ones that differs? I think there are some other no?

**Comment by gsmet** [comment]
I think the solution is perfectly fine. Better use conventions rather than struggling with code. As mentioned by @ia3andy , could you check that we don't have other mismatches in the mojos?

**Comment by mcruzdev** [comment]
> I think the solution is perfectly fine. Better use conventions rather than struggling with code. As mentioned by @ia3andy , could you check that we don't have other mismatches in the mojos?

Hi @gsmet, thank you, I'll take a look at all the goals here.

**Comment by mcruzdev** [comment]
Hi @gsmet  and @ia3andy I just changed field names that are different from `@Parameter#property`.

In `QuarkusBootstrapMojo` I found [this](https://github.com/mcruzdev/quarkus/blob/4c15b43978225e8fd10878414f97d0c036263453/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapMojo.java#L161) when we have:

```java
    @Parameter(property = "quarkusCloseBootstrappedApp")
    private Boolean closeBootstrappedApp;
```

Should I change too?

Another point is that:

In [TrackConfigChangesMojo](https://github.com/mcruzdev/quarkus/blob/4c15b43978225e8fd10878414f97d0c036263453/devtools/maven/src/main/java/io/quarkus/maven/TrackConfigChangesMojo.java#L46) we have some parameters with a extensive `@Parameter#property` name:

```java
    @Parameter(property = "quarkus.track-config-changes.outputFile", required = false)
    File outputFile;
```

In this case, should I add the `name` attribute with value `quarkus.track-config-changes.outputFile`, WTY?

**Comment by mcruzdev** [comment]
I have another question, can't this be a break change for users who use <configuration>?

**Comment by ia3andy** [comment]
>   ```java
>   @Parameter(property = "quarkus.track-config-changes.outputFile", required = false)
>    File outputFile;
> ```

This is to be discussed but IMO we should keep the convention of not using `.` property names if all the other are not using it (so I would switch to another name).

Yes this might be breaking some config so this might require a release note @gsmet ?

**Comment by mcruzdev** [comment]
Hi @ia3andy!

> This is to be discussed but IMO we should keep the convention of not using `.` property names if all the other are not using it (so I would switch to another name).

It is a particularity of this class; if you take a look, all `@Parameter#property` element values are separated by  `.`. This causes a breaking change for the `<configuration>` and `<properties>` sections.

**Comment by gsmet** [comment]
I wonder if we should get rid of this thing entirely and find a way to publish the websites generated by `maven-site-plugin` somehow?

We are trying to reinvent the wheel here and it was fine if it was simple but it's apparently a lot more complex than we thought.

Does it make sense?

Note that I have no idea if this would be feasible/easy...

**Comment by mcruzdev** [comment]
> I wonder if we should get rid of this thing entirely and find a way to publish the websites generated by `maven-site-plugin` somehow?
> 
> We are trying to reinvent the wheel here and it was fine if it was simple but it's apparently a lot more complex than we thought.
> 
> Does it make sense?
> 
> Note that I have no idea if this would be feasible/easy...

I think that is makes sense, I will take a look to see how we can access the property during documentation build.

**Comment by mcruzdev** [comment]
Hi @gsmet and @ia3andy, I added a new way to solve the issue, as mentioned here https://github.com/quarkusio/quarkus/issues/37488#issuecomment-1840239909, there is no way to use Maven API to achieve the issue in the documentation. 

Now, the documentation will be ok with the Maven "User property".

File: docs/src/main/java/io/quarkus/docs/generation/QuarkusMavenPluginDocsGenerator.java [hunk 1]
```diff
@@ -26,6 +27,18 @@ public class QuarkusMavenPluginDocsGenerator {
...
+        if (path.endsWith("docs")) {
+            return path;
+        }
+        return path.resolve("docs");
+    }
+<<<<<<< HEAD
+
+=======
+    
+>>>>>>> a606b074ee654431b75bdcf56c5cc4ca4a4c4584
```
Comment (mcruzdev) [inline docs/src/main/java/io/quarkus/docs/generation/QuarkusMavenPluginDocsGenerator.java:41]:
```suggestion
```

**Comment by mcruzdev** [comment]
hi @gsmet, how are you? Can you review it? Was the only way to solve the issue without breaking the compatibility

**Comment by ia3andy** [comment]
@mcruzdev I think what @gsmet suggested was to use what the plugin is generating by default as doc and then find a way to parse/integrate it in the Quarkus website.

I also wonder what the Maven Site plugin generates, they have the use info in the end no?

**Comment by mcruzdev** [comment]
Hi @ia3andy and @gsmet I think that finally we solve the issue.

**Review by ia3andy — APPROVED** [review]
You didn't back down and found the right solution, very nice!

**Comment by ia3andy** [comment]
@gsmet not sure if we have to backport this to have it in previous versions?

**Pull request merged**